org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/OperatorIds.java

```diff
@@ -36,11 +36,11 @@ public interface OperatorIds {
 	public static final int UNSIGNED_RIGHT_SHIFT= 19;
 	public static final int NumberOfTables = 20;
 
-	public static final int QUESTIONCOLON = 23;
+	public static final int QUESTIONCOLON = 21;
 
-	public static final int NOT_EQUAL = 29;
-	public static final int EQUAL = 30;
-	public static final int INSTANCEOF = 31;
-	public static final int PLUS_PLUS = 32;
-	public static final int MINUS_MINUS = 33;
+	public static final int NOT_EQUAL = 22;
+	public static final int EQUAL = 23;
+	public static final int INSTANCEOF = 24;
+	public static final int PLUS_PLUS = 25;
+	public static final int MINUS_MINUS = 26;
 }
```

Comment:
So, I am nervous here - These numerical assignments date back to almost two decades ago. Are you able to ascertain why the holes were originally left in ? Is that to maintain some kind of mapping ? If so is such mapping no longer relevant ?? What made it irrelevant ?

Reply:
I don't see any reason why there have been those gaps - everything seem to work without those.

Reply:
Please note that there's also https://github.com/eclipse-jdt/eclipse.jdt.core/blob/master/org.eclipse.jdt.core.tests.model/workspace/Compiler/src/org/eclipse/jdt/internal/compiler/ast/OperatorIds.java
If you plan to do some perceived cleanup on the IDs, I suggest to do it in a separate changeset.

Reply:
@szarnekow i don't get what you are trying to say here. that ProfilingASTConvertionTest seems to be unused and totally unrelated to this topic.

Reply:
> Please note that there's also https://github.com/eclipse-jdt/eclipse.jdt.core/blob/master/org.eclipse.jdt.core.tests.model/workspace/Compiler/src/org/eclipse/jdt/internal/compiler/ast/OperatorIds.java If you plan to do some perceived cleanup on the IDs, I suggest to do it in a separate changeset.

AFAICT, this is a snapshot - which need not be kept upto date - @szarnekow Do you think otherwise ?

Reply:
Ok, understood. You're saying that the OperatorIds from the referenced test is only accidentally identical with OperatorIds from the production code. If that's the case, I'm fine with it being no longer the same.

Reply:
> Ok, understood. You're saying that the OperatorIds from the referenced test is only accidentally identical with OperatorIds from the production code. If that's the case, I'm fine with it being no longer the same.

I don't think they are accidentally identical. I think it is a snapshot of the world as it was once upon a time but is not meant to be kept in sync. 

I don't have my desktop right now to confirm. But I would expect many things to be out of sync with main code base.